geopm 1.1.0 - Add support for OpenSUSE 15.1

jcsiadal opened the pull request:
Updated from original pull request -- rebased to latest 2.0 branch.

Updates geopm 1.1 for OpenSUSE 15.1.
Includes changes from geopm source author.

Signed-off-by: jcsiadal <jeremy.c.siadal@intel.com>

jcsiadal committed:
Update geopm to 1.1.0

Signed-off-by: jcsiadal <jeremy.c.siadal@intel.com>

jcsiadal committed:
Update geopm to 1.1.0

Signed-off-by: jcsiadal <jeremy.c.siadal@intel.com>

cmcantalupo commented:
I found an issue with the way that SUSE 15 SP1 is expanding the "python_lib_dir" macro.  On my system is is resolving to python3.4 even though system python is version 3.6.  The solution below works:

https://github.com/cmcantalupo/ohpc/commit/63affab948a06f286a0889dfc8cc1b4b119c36a7#diff-b8e4a72d27eb119bd04fdd9147258aa1R115

jcsiadal commented:
You're right about that. None of the PRs for the HPC_macros file for 2.0 have been merged, so the new distros aren't working. The fix needs to merge into that file. Your method is easier to maintain than the current method of setting Python based on OS version.

cmcantalupo commented:
I added a spec file variable to track the site-package directory in the version here:

https://review.gerrithub.io/c/geopm/geopm/+/477493/

rather than expanding the variable in place in the module file section.  Functionally the same though.

cmcantalupo committed:
Changes geopm spec file to avoid using use python_lib_dir variable directly

- Fixes issue on SUSE 15 SP1 where python_lib_dir expands for python3.4
  but the system python is python3.6.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

jcsiadal committed:
Merge branch 'geopm' into geopm

jcsiadal committed:
Merge pull request #1 from cmcantalupo/geopm

Changes geopm spec file to avoid using use python_lib_dir variable di…

jcsiadal committed:
Minor updates

Signed-off-by: jcsiadal <jeremy.c.siadal@intel.com>

jcsiadal committed:
Rebase to latest 2.0

Signed-off-by: jcsiadal <jeremy.c.siadal@intel.com>
